salt/utils/event.py

```diff
@@ -401,9 +422,10 @@ def _get_event(self, wait, tag, match_func=None):
                 if socks.get(self.sub) != zmq.POLLIN:
                     continue
 
-                # Please do not use non-blocking mode here. Reliability is
-                # more important than pure speed on the event bus.
-                ret = self.get_event_block()
+                if no_block is True:
```

Comment:
If we are going to do a nonblocking call here, we should have the if block encompass the poll() above-- since the poll() blocks.